### ISSUE TITLE
Build fails on Ubuntu 18.04 g++ 7.3.0 because std::tr1 is being used

### ISSUE THREAD

**Issue by jerome-caucat** [ifcquery/ifcplusplus#214]
OS: Ubuntu 18.04
gcc/g++: 7.3.0

Here are the commands I executed:

```bash
$ mkdir build
$ cd build
$ cmake -DBUILD_CONSOLE_APPLICATION=OFF -DBUILD_STATIC_LIBRARY=OFF -DBUILD_VIEWER_APPLICATION=OFF -DCMAKE_BUILD_TYPE=Release ..
$ make
```

I get errors during the `make` command because in [`IfcPlusPlus/src/ifcpp/model/BasicTypes.h#L22`](https://github.com/ifcquery/ifcplusplus/blob/3247e3d8ec009035d85fc986b3b99fc783817eaf/IfcPlusPlus/src/ifcpp/model/BasicTypes.h#L22) the `using std::tr1::shared_ptr; ...` lines are used instead of the `using std::shared_ptr; ...` lines:

```
error: 'std::tr1' has not been declared using std::tr1::shared_ptr;
```

[Full log](https://github.com/ifcquery/ifcplusplus/files/6714725/make.log.zip)

Here is how I patched it:

```diff
$ git diff
diff --git a/IfcPlusPlus/src/ifcpp/model/BasicTypes.h b/IfcPlusPlus/src/ifcpp/model/BasicTypes.h
index c5badcbe..c031e8d3 100644
--- a/IfcPlusPlus/src/ifcpp/model/BasicTypes.h
+++ b/IfcPlusPlus/src/ifcpp/model/BasicTypes.h
@@ -19,16 +19,18 @@ LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OU
 
 #include <memory>
 
-#if _MSC_VER < 1900
+#if defined _MSC_VER
+ #if _MSC_VER < 1900
        using std::tr1::shared_ptr;
        using std::tr1::weak_ptr;
        using std::tr1::dynamic_pointer_cast;
        using std::tr1::make_shared;
-#elif _MSC_VER >= 1900
+ #else
        using std::shared_ptr;
        using std::weak_ptr;
        using std::dynamic_pointer_cast;
        using std::make_shared;
+ #endif
 #elif defined __clang__
        using std::shared_ptr;
        using std::weak_ptr;
```

**Comment by chrisg123** [comment]
I submitted a pull request 18 days ago which resolves this ```Fix build on linux/gcc #212```

**Comment by jerome-caucat** [comment]
Issue fixed in [v1.1](https://github.com/ifcquery/ifcplusplus/releases/tag/1.1), thanks.
